src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -398,6 +404,39 @@ public async Task<Index> CreateIndex(IndexDefinition definition)
             }
         }
 
+        private bool NeedToUpdateIndex(IndexDefinition definition, out Index currentIndex, out bool validateMapReduceCollectionName)
+        {
+            currentIndex = GetIndex(definition.Name);
+            if (currentIndex == null)
+            {
+                // new index
+                validateMapReduceCollectionName = true;
+                return true;
+            }
+
+            var creationOptions = GetIndexCreationOptions(definition, currentIndex, out var _);
+            if (creationOptions != IndexCreationOptions.Noop)
+            {
+                // update/create an index definition
+                validateMapReduceCollectionName = creationOptions == IndexCreationOptions.UpdateWithoutUpdatingCompiledIndex;
+                return true;
+            }
+
+            var replacementIndexName = Constants.Documents.Indexing.SideBySideIndexNamePrefix + definition.Name;
+            if (GetIndex(replacementIndexName) != null)
+            {
+                // original index and side by side index exist
+                // we restored the original index definition
+                // need to delete the side by side index
+                // this is done on each server separately so we need to send the create index command
+                validateMapReduceCollectionName = false;
+                return true;
+            }
+
+            validateMapReduceCollectionName = false;
+            return false;
```

Comment:
This code is seems to have very similar logic as `IndexStore.HandleStaticIndexChange` why do we need it here?

Reply:
There is no need to send a RAFT command if the index didn't change.
HandleStaticIndexChange is after applying the RAFT command.

Reply:
Why should it matter? This is a rare thing, so not much point in optimizing it.

Reply:
Calling IndexCreation.CreateIndexes() on app restart with 100 indexes for example

Reply:
Which would be rare, and would usually be batched anyway, so one call only

Reply:
Talking with @grisha-kotler, we have an issue because we want to make this validation _before_ we accept the command into the cluster.

Reply:
This is too late. The caller for creating the index won't see this error and the current index will be errored (if we changed the definition).

Reply:
I've commented this validation:

https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Server/Documents/Indexes/IndexStore.cs#L383

run `OutputReduceToCollectionTests` - all passed

Reply:
The diffrenece now is that an errored index will be created

Reply:
what about the previous index def at this point?

Reply:
just make a quick test and everything seems to be ok. I got the validation error and previous index def remained unchanged

Reply:
same here. (the replacement index failed to be created which is what we need)

Reply:
hmm.. my approach doesn't seem to be okay. if we skip the above validation we'll end up with DatabaseRecord containing invalid index def :/